master/master.cfg

```diff
@@ -422,6 +425,50 @@ def get_msvc_config_steps(factory, builder_type):
                                extract_fn=save_interesting_env_vars))
 
 
+EMSDK_ENV_VARS = [
+    "EM_CACHE",
+    "EM_CONFIG",
+    "EMSDK",
+    "EMSDK_NODE",
+    "EMSDK_PYTHON",
+    "PATH",
+]
+
+def get_emsdk_config_steps(factory, builder_type):
+    if not builder_type.handles_wasm():
+        print("WARNING: should not call get_emsdk_config_steps() for builds that cannot handle WASM")
+        return
+
+    def save_interesting_env_vars(rc, stdout, stderr):
+        d = {}
+        for line in stdout.split('\n'):
+            match = re.match("^([a-zA-Z0-9_-]+) = (.*)$", line.strip())
+            if match:
+                key = match.group(1).upper()
+                value = match.group(2)
+                if key in [EMSDK_ENV_VARS]:
+                    d[key] = value
+        print("emsdk env: ", d)
+        return {'emsdk_env': d}
+
+    factory.addStep(
+        SetPropertyFromCommand(name='Run emsdk_env',
+                               description='Run emsdk_env',
+                               locks=[performance_lock.access('counting')],
+                               haltOnFailure=True,
+                               env=Property('env'),
+                               command="source ${EMSDK}/emsdk_env.sh",
+                               extract_fn=save_interesting_env_vars))
+
+def extend_with_emsdk_config(env):
+    @renderer
+    def render(props):
+        emsdk_env = props.getProperty('emsdk_env', default={})
+        env.update(emsdk_env)
+        return env
+
+    return render
```

Comment:
I'm not sure if this will work. I would have written it like:
```
def merge_renderable(base, extn):
    @renderer
    @defer.inlineCallbacks
    def render(props):
        base = yield props.render(base)
        extn = yield props.render(extn)
        base.update(extn)
        return base

    return render
```

Reply:
Shamelessly plagiarized

Reply:
I just hope that's correct!